packages/tools/devtools/devtools-core/src/data-visualization/DataEditing.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISharedObject } from "@fluidframework/shared-object-base";
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { EditType, HasFluidObjectId } from "../CommonInterfaces";
+
+/**
+ * Applies an edit to {@link @fluidframework/shared-object-base#ISharedObject}.
+ * @param sharedObject - The {@link @fluidframework/shared-object-base#ISharedObject} whose data will be edited.
+ * @param edit - Describes what changes will be made using {@link Edit}.
+ *
+ * @internal
+ */
+export type EditSharedObject = (sharedObject: ISharedObject, edit: Edit) => Promise<void>;
+
+/**
+ * Interface to contain information necesary for an edit
+ * @internal
+ */
+export interface Edit {
+	/**
+	 * Type contains the {@link EditType} of the edit being preformed
+	 */
+	type?: EditType;
```

Comment:
Sorry, this didn't occur to me earlier: we can enhance forward / backward compatibility here by updating the type information a bit. 

```typescript
/**
 * Type contains the {@link EditType} of the edit being preformed.
 *
 * @remarks This is generally expected to be of type `EditType`. `string` is supported strictly for forward / backward compatibility.
 */
type?: EditType | string;
```

Since our enum is typed as `string`, if we add or remove values from it, old / new values will still be supported. Additionally, it signals to consuming code that it needs to gracefully handle this being an unexpected value.

Reply:
Can you add the remarks comment block here as well? I think it's important to call out _why_ we allow string.